packages/payload/src/fields/validations.ts

```diff
@@ -587,7 +587,12 @@ const validateFilterOptions: Validate<
         return true
       }
 
-      return options[collection].indexOf(requestedID) === -1
+      const idType =
+        payload.collections[collection]?.customIDType || payload?.db?.defaultIDType || 'text'
+
+      // Cast the ID to the correct type
+      const id = idType === 'number' ? Number(requestedID) : String(requestedID)
```

Comment:
We could simplify the comparison by always comparing strings. Cast both the options IDs and the input value to string. We will have simpler validation with less code to manage if we do it right.